### ISSUE TITLE
Chart replication fails when the name or tag includes uppercase letters

### ISSUE THREAD

**Issue by alinbalutoiu** [goharbor/harbor#13187]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
Expecting the helm charts replication to work without issues if the helm charts exist.
Actual behavior: replicating helm charts fails.

**Steps to reproduce the problem:**
- Setup Harbor `v2.1.0`
- Upload a helm chart to another Harbor registry (tested with `v1.9.2`) that includes uppercase letters, example: `library/Test-Chart:dev`
- Setup replication of the helm charts between the Harbor `v1.9.2` and Harbor `v2.1.0`
- Trigger the replication

**Versions:**
Please specify the versions of following systems.

- harbor version: [2.1.0]
- docker engine version: [N/A]
- docker-compose version: [N/A]

**Additional context:**
The actual error:
```
[ERROR] [/replication/transfer/chart/transfer.go:156]: failed to download the chart library/Test-Chart:dev: failed to download the chart https://<redacted>/chartrepo/library/charts/test-chart-dev.tgz: 404 {"error":"object not found"}
```

Issue from the CLI:
```bash
$ curl https://<redacted>/chartrepo/library/charts/test-chart-dev.tgz
{"error":"object not found"}
$ curl https://<redacted>/chartrepo/library/charts/Test-Chart-dev.tgz
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.
```

Possibly coming from:
https://github.com/goharbor/harbor/blob/v2.1.0/src/replication/transfer/chart/transfer.go#L154
https://github.com/goharbor/harbor/blob/v2.1.0/src/replication/adapter/harbor/base/chart_registry.go#L154

**Comment by steven-zou** [comment]
@mmpei Please take a look at this issue, thanks!

**Comment by mmpei** [comment]
OK, it seems that it's case sensitive for chart's name. i will check.

**Comment by chrishoerl** [comment]
I have the same problem. Harbor v2.1.0
This was already opened here but closed because of staleness.
#9798 

Please fix.

**Comment by mmpei** [comment]
fixed in release 2.0.4 2.1.2 and futures.
